src/pages/workspaces/workspace/analysis/modals/AnalysisDuplicator.test.ts

```diff
@@ -0,0 +1,282 @@
+    const analysisMock: Partial<GoogleStorageContract['analysis']> = jest.fn(() => ({
+      rename
+    }))
+    const googleStorageMock: Partial<GoogleStorageContract> = ({
+      analysis: analysisMock as GoogleStorageContract['analysis']
+    })
+    asMockedFn(GoogleStorage).mockImplementation(() => googleStorageMock as GoogleStorageContract)
+
+    // Act
+    render(h(AnalysisDuplicator, { ...defaultModalProps, destroyOld: true }))
+
+    // Assert
+    const inputText = 'newName'
+    const input = await screen.findByLabelText(/New Name/)
+    await userEvent.type(input, inputText)
+    const button = screen.getByText('Rename Analysis')
+    await userEvent.click(button)
+
+    expect(analysisMock).toHaveBeenCalledWith(defaultGoogleWorkspace.workspace.googleProject, defaultGoogleWorkspace.workspace.bucketName, defaultModalProps.printName, defaultModalProps.toolLabel)
+    expect(rename).toHaveBeenCalledWith(inputText)
+  })
+
+  it('copies for an azure workspace correctly', async () => {
+    // Arrange
+    const copy = jest.fn()
+    const analysisMock: Partial<AzureStorageContract['blob']> = jest.fn(() => ({
+      copy
+    }))
+
+    const azureStorageMock: Partial<AzureStorageContract> = ({
+      blob: analysisMock as AzureStorageContract['blob']
+    })
+    asMockedFn(AzureStorage).mockImplementation(() => azureStorageMock as AzureStorageContract)
+
+    // Act
+    render(h(AnalysisDuplicator, { ...defaultModalProps, workspaceInfo: defaultAzureWorkspace.workspace }))
+
+    // Assert
+    const input = screen.getByLabelText(/New Name/)
+    const inputText = 'newName'
+    await userEvent.type(input, inputText)
+    const button = screen.getByText('Copy Analysis')
+    await userEvent.click(button)
+
+    expect(analysisMock).toHaveBeenCalledWith(defaultAzureWorkspace.workspace.workspaceId, defaultModalProps.printName)
+    expect(copy).toHaveBeenCalledWith(inputText)
+  })
+
+  it('renames for an azure workspace correctly', async () => {
+    // Arrange
+    const rename = jest.fn()
+    const analysisMock: Partial<AzureStorageContract['blob']> = jest.fn(() => ({
+      rename
+    }))
+
+    const azureStorageMock: Partial<AzureStorageContract> = ({
+      blob: analysisMock as AzureStorageContract['blob']
+    })
+    asMockedFn(AzureStorage).mockImplementation(() => azureStorageMock as AzureStorageContract)
+
+    // Act
+    render(h(AnalysisDuplicator, { ...defaultModalProps, workspaceInfo: defaultAzureWorkspace.workspace, destroyOld: true }))
+
+    // Assert
+    const inputText = 'newName'
+    const input = screen.getByLabelText(/New Name/)
+    await userEvent.type(input, inputText)
+    const button = screen.getByText('Rename Analysis')
+    await userEvent.click(button)
+
+    expect(analysisMock).toHaveBeenCalledWith(defaultAzureWorkspace.workspace.workspaceId, defaultModalProps.printName)
+    expect(rename).toHaveBeenCalledWith(inputText)
+  })
+
+  // TODO: testing curried error handling needs investigation...
```

Comment:
let's pair on this soon.

Reply:
Ok. Seems this could be a separate PR based on some previous conversations we had about the complexity. Let me know.

Reply:
I paired with Justin on this test recently and we got it fixed up to where it just needed some further follow-up on less serious aspects. So, fingers crossed.